### ISSUE TITLE
waitForAttach only takes IFluidDataStoreRuntime. Could it get IFluidDataStoreContext as well?

### ISSUE THREAD

**Issue by cosmincatrinescu** [microsoft/FluidFramework#4229]
It looks like we can make waitForAttach take a union type of IFluidDataStoreRuntime | IFluidDataStoreContext, for everything we use in that method they are compatible

**Comment by anthony-murphy** [comment]
Just need to update this method signature to take the union `IFluidDataStoreRuntime | IFluidDataStoreContext`
https://github.com/microsoft/FluidFramework/blob/81795eedd5c562646a62a70c8b8996dcf3e86746/packages/framework/aqueduct/src/utils/attachUtils.ts#L9

**Comment by curtisman** [comment]
I am wondering when you will get an IFluidDataStoreContext but not the IFluidDataStoreRuntime?

**Comment by curtisman** [comment]
@cosmincatrinescu can you tell me more about the situation where you get an IFluidDataStoreContext but not the IFluidDataStoreRuntime?

**Comment by anthony-murphy** [comment]
I don't have a stong viewpoint here, but like the goal of minimizing context. not sure what the implications of the in the FFX code base are. Maybe this is a good case to explore that work.

**Comment by cosmincatrinescu** [comment]
I guess it's fair to say you had to have gotten the runtime in order to have the context. If access to IFluidDataStoreContext would not have been available, I guess the 2-3 levels deep stack I'm looking at would have passed the IFluidDataStoreRuntime around instead of the context. I can add a task on FFX to switch to passing IFluidDataStoreRuntime around.

**Comment by cosmincatrinescu** [comment]
@curtisman I peeked at the linked PR, not sure it's in fact related to this issue? 

At any rate, @curtisman @anthony-murphy, while looking into using IFluidDataStoreRuntime instead of IFluidDataStoreContext in the FFX code, I learned more about why we used the context. The reason this is passed around is to get to the context.containerRuntime and then to context.containerRuntime.getAbsoluteUrl. I don't seem to have a way to get to the containerRuntime from IFluidDataStoreRuntime....is that correct or is there an indirection that can take me there?
